base_layer/contacts/src/contacts_service/storage/types/contacts.rs

```diff
@@ -0,0 +1,184 @@
+        address: &[u8],
+        conn: &mut SqliteConnection,
+    ) -> Result<ContactSql, ContactsServiceStorageError> {
+        Ok(contacts::table
+            .filter(contacts::address.eq(address))
+            .first::<ContactSql>(conn)?)
+    }
+
+    /// Find a particular Contact by their node ID, if it exists
+    pub fn find_by_node_id(
+        node_id: &[u8],
+        conn: &mut SqliteConnection,
+    ) -> Result<ContactSql, ContactsServiceStorageError> {
+        Ok(contacts::table
+            .filter(contacts::node_id.eq(node_id))
+            .first::<ContactSql>(conn)?)
+    }
+
+    /// Find a particular Contact by their address, and update it if it exists, returning the affected record
+    pub fn find_by_address_and_update(
+        conn: &mut SqliteConnection,
+        address: &[u8],
+        updated_contact: UpdateContact,
+    ) -> Result<ContactSql, ContactsServiceStorageError> {
+        // Note: `get_result` not implemented for SQLite
+        diesel::update(contacts::table.filter(contacts::address.eq(address)))
+            .set(updated_contact)
+            .execute(conn)
+            .num_rows_affected_or_not_found(1)?;
+        ContactSql::find_by_address(address, conn)
+    }
+
+    /// Find a particular Contact by their address, and delete it if it exists, returning the affected record
+    pub fn find_by_address_and_delete(
+        conn: &mut SqliteConnection,
+        address: &[u8],
+    ) -> Result<ContactSql, ContactsServiceStorageError> {
+        // Note: `get_result` not implemented for SQLite
+        let contact = ContactSql::find_by_address(address, conn)?;
+        if diesel::delete(contacts::table.filter(contacts::address.eq(address))).execute(conn)? == 0 {
+            return Err(ContactsServiceStorageError::ValuesNotFound);
+        }
+        Ok(contact)
+    }
+
+    /// Find a particular Contact by their node ID, and update it if it exists, returning the affected record
+    pub fn find_by_node_id_and_update(
+        conn: &mut SqliteConnection,
+        node_id: &[u8],
+        updated_contact: UpdateContact,
+    ) -> Result<ContactSql, ContactsServiceStorageError> {
+        // Note: `get_result` not implemented for SQLite
+        diesel::update(contacts::table.filter(contacts::node_id.eq(node_id)))
+            .set(updated_contact)
+            .execute(conn)
+            .num_rows_affected_or_not_found(1)?;
+        ContactSql::find_by_node_id(node_id, conn)
+    }
+
+    /// Find a particular Contact by their node ID, and delete it if it exists, returning the affected record
+    pub fn find_by_node_id_and_delete(
+        conn: &mut SqliteConnection,
+        node_id: &[u8],
+    ) -> Result<ContactSql, ContactsServiceStorageError> {
+        // Note: `get_result` not implemented for SQLite
+        let contact = ContactSql::find_by_node_id(node_id, conn)?;
+        if diesel::delete(contacts::table.filter(contacts::node_id.eq(node_id))).execute(conn)? == 0 {
```

Comment:
You cant use `.num_rows_affected_or_not_found(1)?;` like above?

Reply:
This code was moved over from its previous location in the refactor not written new. I can try out the recommendation though.